lib/hanami/routing/middleware/parser.rb

```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+
+require 'hanami/utils/hash'
+
+module Hanami
+  module Routing
```

Comment:
My suggestion is to provide a single Ruby namespace for all the Hanami middleware, despite the gem they are distributed with. So for instance, `Hanami::Middleware::BodyParsers` is distributed by `hanami-router`, and another (fictional) middleware like `Hanami::Middleware::DoSomethingWithAssets` can be distributed by `hanami-assets`.

In this way both `hanami` and final users know for sure that `Hanami::Middleware` hosts all the Rack middleware.

Reply:
I really like this idea I will move the code to that namespace